src/python/pants/backend/java/dependency_inference/rules.py

```diff
@@ -73,9 +108,26 @@ async def infer_java_dependencies_via_imports(
 
         types.update(maybe_qualify_types)
 
+    # Resolve the export types into (probable) types:
+    # First produce a map of known consumed unqualified types to possible qualified names
+    consumed_type_mapping: dict[str, set[str]] = defaultdict(set)
+    for typ in types:
+        unqualified = typ.rpartition(".")[2]  # `"org.foo.Java"` -> `("org.foo", ".", "Java")`
```

Comment:
What if the symbol has less than 3 components? Maybe use `-1` to index here?

Reply:
@tdyas `rpartition` splits into `(left, sep, right)` so `boop.Foo` will give `("boop", ".", "Foo")`.

Reply:
A package isn't required though: see #13536.

Reply:
So we need to add support to export types in the unnamed package if the compilation unit is in the unnamed package?

Reply:
Unsure of the implications... but we shouldn't crash, heh.

Reply:
👍